apps/shipments/serializers/tags.py

```diff
@@ -0,0 +1,53 @@
+"""
+Copyright 2020 ShipChain, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from rest_framework_json_api import serializers
+
+from apps.permissions import get_owner_id
+from ..models import ShipmentTag
+
+
+class ShipmentTagSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = ShipmentTag
+        fields = '__all__'
+
+
+class ShipmentTagCreateSerializer(ShipmentTagSerializer):
+
+    class Meta:
+        model = ShipmentTag
+        if settings.PROFILES_ENABLED:
+            exclude = ('shipment', 'owner_id', )
+        else:
+            exclude = ('shipment', )
+
+    def validate(self, attrs):
+
+        attrs['shipment_id'] = self.context['view'].kwargs.get('shipment_pk')
+        if settings.PROFILES_ENABLED:
+            attrs['owner_id'] = get_owner_id(self.context['request'])
+
+        try:
+            ShipmentTag(**attrs).validate_unique()
```

Comment:
In a previous iteration of this feature, I did use this validator however I had to tweak it usage because it didn't perform on object creation but rather on update. See here: https://github.com/ShipChain/transmission/pull/176/commits/879b6526635992df26c74d271d20b4e12e30f503#diff-53d6f147459043773e9dc64ef1f04146R32
If you are okay with such implementation I can move it here.

Reply:
Are you positive? That diff does not show the validator being used as documented (defined in the Meta section of the serializer). I also do not see anything in the DRF docs about validators not working on create. Maybe try this approach one more time and if it doesn't seem to be working, myself or @mlclay can take a look at it and see if we can figure out what's wrong with DRF. As far as I know, the views call serializer.is_valid before perform_create(serializer), so all of these validators should be evaluated.